.circleci/config.yml

```diff
@@ -1,23 +1,402 @@
+      SENTRY_ENABLED: false
+      AWS_FOLDER_NAME: ''
+      MATTERMOST_WEBHOOK_URL: ''
+    steps:
+      - build-android
+      - persist
+      - save:
+          filename: "Mattermost.apk"
+
+  build-android-pr:
+    executor: android
+    environment:
+      BRANCH_TO_BUILD: ${CIRCLE_BRANCH}
+    steps:
+      - build-android
+      - persist
+      - save:
+          filename: "Mattermost_Beta.apk"
+
+  build-ios-beta:
+    executor: ios
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost_Beta.ipa"
+
+  build-ios-release:
+    executor: ios
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost.ipa"
+
+  build-ios-pr:
+    executor: ios
+    environment:
+      BRANCH_TO_BUILD: ${CIRCLE_BRANCH}
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost_Beta.ipa"
+
+  deploy-android-release:
+    executor:
+      name: android
+      resource_class: medium
+    steps:
+      - deploy-android:
+          apk_path: Mattermost.apk
+
+  deploy-android-beta:
+    executor:
+      name: android
+      resource_class: medium
+    steps:
+      - deploy-android:
+          apk_path: Mattermost_Beta.apk
+
+  deploy-ios-release:
+    executor: ios
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost.ipa
+
+  deploy-ios-beta:
+    executor: ios
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost_Beta.ipa
 
 workflows:
   version: 2
-  pr-test:
+  build:
     jobs:
       - test
+
+      - build-android-release:
+          context: mattermost-mobile-android-release
+          requires:
+            - test
+          filters:
+            branches:
+              only:
+                - /^build-[0-9]*$/
+                - /^build-(android).*$/
+                - ci # remove
+      - deploy-android-release:
+          context: mattermost-mobile-android-release
+          requires:
+            - build-android-release
+          filters:
+            branches:
+              only:
+                - /^build-[0-9]*$/
+                - /^build-(android).*$/
+
+      - build-android-beta:
+          context: mattermost-mobile-android-beta
+          requires:
+            - test
+          filters:
+            branches:
+              only:
+                - /^build-[0-9]*$/
+                - /^build-(android).*$/
+                - ci # remove
+      - deploy-android-beta:
+          context: mattermost-mobile-android-beta
+          requires:
+            - build-android-beta
+          filters:
+            branches:
+              only:
+                - /^build-[0-9]*$/
+                - /^build-(android).*$/
+
+      - build-ios-release:
+          context: mattermost-mobile-ios-release
+          requires:
+            - test
+          filters:
+            branches:
+              only:
+                - /^build-[0-9]*$/
+                - /^build-(ios).*$/
```

Comment:
why having `(ios)` parens are not needed as you are not using any other operator